source/mainnet/smart-contracts/guides/factory-pattern.rst

```diff
@@ -0,0 +1,392 @@
+.. _factory-pattern:
+
+=======================================
+Using the factory pattern on Concordium
+=======================================
+
+.. Warning::
+
+    This guide makes use of features that are first available in protocol version 7,
+    namely getting the contract name and module reference of a smart contract instance
+    from a smart contract.
+    Before the protocol upgrade to version 7, these features will not work.
```

Comment:
Could we link to a page where people can check when version 7 will be expected to be live (or give a tentative date here e.g. around summer).

Reply:
We don't have that yet.